sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -0,0 +1,594 @@
+        /// </summary>
+        ///
+        public string EventHubName => _producer.EventHubName;
+
+        /// <summary>
+        ///   A unique name to identify the buffered producer.
+        /// </summary>
+        ///
+        public string Identifier => _producer.Identifier;
+
+        /// <summary>
+        ///   The total number of events that are currently buffered and waiting to be published, across all partitions.
+        /// </summary>
+        ///
+        public virtual int TotalBufferedEventCount { get; private set; }
+
+        /// <summary>
+        ///   <c>true</c> if the producer has been closed <c>false</c> otherwise.
+        /// </summary>
+        ///
+        public bool IsClosed => _isClosed;
+
+        /// <summary>
+        ///    Invoked after each batch of events has been successfully published to the Event Hub, this
+        ///    handler is optional and is intended to provide notifications for interested listeners.
+        /// </summary>
+        ///
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0002:DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.", Justification = "Guidance does not apply; this is an event.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0003:DO make service methods virtual.", Justification = "This member follows the standard .NET event pattern; override via the associated On<<EVENT>> method.")]
+        public event Func<SendEventBatchSuccessEventArgs, Task> SendEventBatchSuccessAsync
+        {
+            add
+            {
+                Argument.AssertNotNull(value, nameof(SendEventBatchSuccessAsync));
+
+                if (_sendSucceeded != default)
+                {
+                    throw new NotSupportedException(Resources.HandlerHasAlreadyBeenAssigned);
+                }
+                _sendSucceeded = value;
+            }
+
+            remove
+            {
+                Argument.AssertNotNull(value, nameof(SendEventBatchSuccessAsync));
+
+                if (_isStarted)
+                {
+                    throw new NotSupportedException(Resources.HandlerHasAlreadyBeenAssigned);
+                }
+
+                if (_sendSucceeded != value)
+                {
+                    throw new ArgumentException(Resources.HandlerHasNotBeenAssigned);
+                }
+                _sendSucceeded = default;
+            }
+        }
+
+        /// <summary>The producer to use to send events to the Event Hub.</summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2213:Disposable fields should be disposed", Justification = "It is being disposed but it must be in CloseAsync so that dispose can match the IAsyncDisposable signature.")]
+        private readonly EventHubProducerClient _producer;
```

Comment:
Can we please move the class members right after the static member and before the event declarations?